lib/Doctrine/Common/Collections/Collection.php

```diff
@@ -32,6 +37,9 @@ interface Collection extends Countable, IteratorAggregate, ArrayAccess
      * @param mixed $element The element to add.
      *
      * @return bool Always TRUE.
+     *
+     * @psalm-param T $element
+     * @psalm-return true
```

Comment:
Is psalm-return necessary when it’s the same as the return annotation?

Reply:
It's not the same, `@psalm-return` here specifies narrower type (`true`) as opposed to `bool` specified by `@return`.

Reply:
I believe we should just change the return type itself then.

Reply:
Other tools may not support literal boolean types like `true` and `false`.

Reply:
According to the [FIG specification](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#appendix-a-types) it should indeed be possible to use literal types in the normal PHPDoc. Hence, I agree that we could remove the psalm annotation and modify the other annotation to state `true` as the return type. If other tools don't support that, they seemingly don't implement the spec correctly as far as I can tell.

Reply:
Ah, I wasn't aware that was a part of PSR-5 (even though it's not a formally ratified standard, it's still widely used). Then I guess it'd be ok to replace it with `@return true`.